### PR TITLE
[Tables] Add AAD sample

### PR DESCRIPTION
**Pull request by YalinLi0312**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/33712

The AAD sample is missing in our doc: https://learn.microsoft.com/en-us/python/api/overview/azure/data-tables-readme?view=azure-python#types-of-credentials

### CODE REVIEW

File: sdk/tables/azure-data-tables/README.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ The `credential` parameter may be provided in a number of different forms, depen
 * Shared Key
 * Connection String
 * Shared Access Signature Token
+* TokenCredential(AAD)
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/README.md:63]: I think this is for Storage only - I don't think it's supported by Cosmos, so we should make a note of that.

File: sdk/tables/azure-data-tables/README.md [hunk 2]
```diff
@@ -69,15 +70,19 @@ az storage account keys list -g MyResourceGroup -n MyStorageAccount
...
 ```python
-from azure.core.credentials import AzureNamedKeyCredential
 from azure.data.tables import TableServiceClient
+from azure.core.credentials import AzureNamedKeyCredential
 
-credential = AzureNamedKeyCredential("my_account_name", "my_access_key")
-
-service = TableServiceClient(endpoint="https://<my_account_name>.table.core.windows.net", credential=credential)
+credential = AzureNamedKeyCredential(self.account_name, self.access_key)
+table_service_client = TableServiceClient(endpoint=self.endpoint, credential=credential)
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/README.md:81]: Having things like `self.endpoint` in the code is really unhelpful for a readme.... even though I understand this is for testing.....

File: sdk/tables/azure-data-tables/samples/sample_authentication.py [hunk 3]
```diff
@@ -40,56 +45,63 @@ def __init__(self):
         self.connection_string = f"DefaultEndpointsProtocol=https;AccountName={self.account_name};AccountKey={self.access_key};EndpointSuffix={self.endpoint_suffix}"
 
     def authentication_by_connection_string(self):
-        # Instantiate a TableServiceClient using a connection string
+        print("Instantiate a TableServiceClient using a connection string")
         # [START auth_from_connection_string]
         from azure.data.tables import TableServiceClient
 
-        with TableServiceClient.from_connection_string(conn_str=self.connection_string) as table_service:
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/samples/sample_authentication.py:47]: Why did we remove the `with` syntax?

Comment (YalinLi0312) [reply sdk/tables/azure-data-tables/samples/sample_authentication.py:47]: Updated to use `with`.

File: sdk/tables/azure-data-tables/README.md (hunk 2 above)
Comment (YalinLi0312) [reply sdk/tables/azure-data-tables/README.md]: Changed back. I don't know if ci has snippet consistency check, if it has, we have to remove the snippet annotations to keep the samples in README clear.

File: sdk/tables/azure-data-tables/README.md (hunk 1 above)
Comment (YalinLi0312) [reply sdk/tables/azure-data-tables/README.md]: Added.

**Review by annatisch — APPROVED** [review]

**Pull request merged**